.github/workflows/update_version.yml

```diff
@@ -49,29 +51,49 @@ jobs:
           pip install wheel
           pip install --editable ".[all]"
 
+      # update all the version strings and add CHANGELOG headings
       - name: Update version
         run: python scripts/update_version.py
 
+      # create a new version branch for rc0 release and commit
       - uses: EndBug/add-and-commit@v9
         if: github.event_name == 'schedule'
         with:
           message: 'Bump to ${{ env.VERSION }}'
           new_branch: '${{ env.NON_RC_VERSION }}'
 
+      # use the already created release branch for rcX + final releases
+      # and commit
       - uses: EndBug/add-and-commit@v9
         if: github.event_name == 'workflow_dispatch'
         with:
           message: 'Bump to ${{ env.VERSION }}'
 
-      - name: Make a PR from ${{ env.NON_RC_VERSION }} to develop
-        uses: repo-sync/pull-request@v2
+      # checkout to develop for updating versions in the same
+      - uses: actions/checkout@v4
         with:
-          source_branch: '${{ env.NON_RC_VERSION }}'
-          destination_branch: "develop"
-          pr_title: "Sync ${{ env.NON_RC_VERSION }} and develop"
-          pr_body: "**Merge as soon as possible to avoid potential conflicts.**"
-          github_token: ${{ secrets.GITHUB_TOKEN }}
+          ref: 'develop'
+
+      # update all the version strings
+      - name: Update version
+        if: github.event_name == 'workflow_dispatch'
+        run: python scripts/update_version.py
+
+      # create a pull request updating versions in develop
+      - name: Create Pull Request
+        id: version_pr
+        uses: peter-evans/create-pull-request@v3
+        with:
```

Comment:
Now we don't "sync" the release and develop branches, instead we just update the versions in both the branches independently. This saves us from the conflicts and duplicating the cherry-picked commits.